File: isma-hsm/src/main/java/ru/nstu/isma/core/hsm/exp/EXPOperand.java
package ru.nstu.isma.core.hsm.exp;

import ru.nstu.isma.core.hsm.var.HMVariable;

import java.io.Serializable;

/**
 * Created by Bessonov Alex
 * Date: 29.11.13
 * Time: 0:57
 */
public class EXPOperand extends EXPToken implements Serializable {
    protected HMVariable variable;

    public EXPOperand() {
    }

    public EXPOperand(HMVariable variable) {
        this.variable = variable;
    }

    public HMVariable getVariable() {
        return variable;
    }

    public void setVariable(HMVariable variable) {
        this.variable = variable;
    }

    @Override
    public String toString() {
        return variable.toString();
    }
}

File: isma-next-hsm/src/main/kotlin/ru/nstu/isma/hsm/exp/EXPOperand.kt
package ru.nstu.isma.hsm.exp

import ru.nstu.isma.hsm.`var`.HMVariable
import java.io.Serializable

/**
 * Created by Bessonov Alex
 * Date: 29.11.13
 * Time: 0:57
 */
open class EXPOperand : EXPToken, Serializable {
    var variable: HMVariable? = null

    constructor() {}
    constructor(variable: HMVariable?) {
        this.variable = variable
    }

    override fun toString(): String {
        return variable.toString()
    }
}